staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -608,3 +646,63 @@ func (c *$.type|privatePlural$) Patch(ctx context.Context, name string, pt $.Pat
 	return
 }
 `
+
+var applyTemplate = `
+// Apply takes the given apply declarative configuration, applies it and returns the applied $.resultType|private$.
+func (c *$.type|privatePlural$) Apply(ctx context.Context, $.inputType|private$ *$.applyConfig|raw$, opts $.ApplyOptions|raw$) (result *$.resultType|raw$, err error) {
```

Comment:
Would it be worth it to check if the field manager is set or would it be fine to defer it to the server?

Reply:
We only infer it from Update, never from patch or apply. So this is accurate.

I'd like to leave enforcement up to the server. My thinking here is (1) the request is going to fail anyway, so there is no additional risk and (2) While not a big deal in this case, I'd like to bias towards keep logic on the server wherever possible to keep our clients thin.